### ISSUE TITLE
Create tool_type to identify similar tools

### ISSUE THREAD

**Issue by pkoenig10** [sc0v/binder-app#100]
This needs to be done before #95 and #101.

**Comment by purelogiq** [comment]
I've been thinking about this issue recently. Is there any value in having "tool types" in addition to "tool name"? Checkout can look at `binder-app/db/tool_seed.rb` to see what I mean.

Maybe this problem can be solved by just thinking differently about tools. We want to be able to identify an individual tool as well as a category of tools. I feel that a the "name" field in the tool model should be its type (and misuses of that field should be changed). The tool "description" can be a more detailed version of its "name". For example lets say you have two tools that are different but can sometimes be used for the same purpose:

1) Craftsman Red 1/8in bit Powerdrill
2) Panasonic Blue Versitile Powerdrill

Under the current system we could just have:
{barcode: 1, name: "Powerdrill", description: "Craftsman Red 1/8in bit Powerdrill"}
{barcode: 2, name: "Powerdrill", description: "Panasonic Blue Versitile Powerdrill"}

I proposed we rename the "name" filed to "type" to resolve this issue and the description field can be used more like a "full name and other info field". This allows the software to identify a unique tool (by barcode), both the software and a human to identify a category of tool (the type field), and also allows a human to identify a unique tool (the description).

Because this issue is highly related to #101 I want to clarify that the solution for that issue will be able to create a waitlist for a type of tool as well as a specific individual tool (e.g. for someone that really needs exactly the Panasonic powerdrill).

(I understand renaming name to type isn't the perfectly normalized database schema but in this case the extra model for tool_type would be an unnecessary extra step for users when adding a new tool)

**Comment by purelogiq** [comment]
Or alternatively move the name field to the new tool type model so that there isn't the confusion between seeing both the tool name and tool type have the same info, for example:

| barcode | name      | description    | type      |
|---------|-----------|----------------|-----------|
| 2       | SCC Radio | Motorola CP200 | SCC Radio |

**Comment by minigeek** [comment]
I think that's fine, I would add a tool_type model with just a unique "name" column (and maybe a description, up to you) to enforce consistency (something I thought about doing before but never accomplished) and use "type" as the foreign key to that model

**Comment by purelogiq** [comment]
Just be sure do you think removing the name field in tool is ok, since it would probably be duplicate of the type name?

**Comment by minigeek** [comment]
Yeah, that's pretty much how we were using it before regardless, as you
correctly pointed out.

**Comment by purelogiq** [comment]
Will do

**Comment by pkoenig10** [comment]
I agree that we definitely want a `tool_type` model to ensure consistent naming.  And this can certainly replace name in the `tool` model.  We will just use the description for more specific information.

**Comment by arakla** [comment]
Just a side note, this will probably affect how binder displays certain
tools such as hardhats and radios, but it should be a easy fix.

**Comment by purelogiq** [comment]
Yea, I'll try to make this a comprehensive change and not just add the models.
